src/Product.jl

```diff
@@ -0,0 +1,85 @@
+using LinearAlgebra
+using Graphs
+
+struct Product <: AbstractAnsatz
+    tn::Ansatz
+end
+
+Ansatz(tn::Product) = tn.tn
+
+Base.copy(x::Product) = Product(copy(Ansatz(x)))
+Base.similar(x::Product) = Product(similar(Ansatz(x)))
+Base.zero(x::Product) = Product(zero(Ansatz(x)))
+
+function Product(arrays::AbstractArray{<:AbstractVector})
+    n = length(arrays)
+    gen = IndexCounter()
+    symbols = map(arrays) do _
+        nextindex!(gen)
+    end
+    _tensors = map(eachindex(arrays)) do i
+        Tensor(arrays[i], [symbols[i]])
+    end
+
+    sitemap = Dict(Site(i) => symbols[i] for i in eachindex(arrays))
+    qtn = Quantum(TensorNetwork(_tensors), sitemap)
+    graph = Graph(n)
+    mapping = BijectiveIdDict{Site,Int}(Pair{Site,Int}[site => i for (i, site) in enumerate(lanes(qtn))])
+    lattice = Lattice(mapping, graph)
+    ansatz = Ansatz(qtn, lattice)
+    return Product(ansatz)
+end
+
+function Product(arrays::AbstractArray{<:AbstractMatrix})
+    n = length(arrays)
+    gen = IndexCounter()
+    symbols = map(arrays) do _
+        (nextindex!(gen), nextindex!(gen))
+    end
+    _tensors = map(eachindex(arrays)) do i
+        Tensor(arrays[i], [symbols[i][1], symbols[i][2]])
+    end
+
+    sitemap = merge!(
+        Dict(Site(i; dual=true) => symbols[i][1] for i in eachindex(arrays)),
+        Dict(Site(i) => symbols[i][2] for i in eachindex(arrays)),
+    )
+    qtn = Quantum(TensorNetwork(_tensors), sitemap)
+    graph = Graph(n)
+    mapping = BijectiveIdDict{Site,Int}(Pair{Site,Int}[site => i for (i, site) in enumerate(lanes(qtn))])
+    lattice = Lattice(mapping, graph)
+    ansatz = Ansatz(qtn, lattice)
+    return Product(ansatz)
+end
+
+LinearAlgebra.norm(tn::Product, p::Real=2) = LinearAlgebra.norm(socket(tn), tn, p)
+function LinearAlgebra.norm(::Union{State,Operator}, tn::Product, p::Real)
+    return mapreduce(*, tensors(tn)) do tensor
+        norm(tensor, p)
+    end^(1//p)
+end
+
+LinearAlgebra.opnorm(tn::Product, p::Real=2) = LinearAlgebra.opnorm(socket(tn), tn, p)
+function LinearAlgebra.opnorm(::Operator, tn::Product, p::Real)
+    return mapreduce(*, tensors(tn)) do tensor
+        opnorm(parent(tensor), p)
+    end^(1//p)
+end
+
+LinearAlgebra.normalize!(tn::Product, p::Real=2) = LinearAlgebra.normalize!(socket(tn), tn, p)
+function LinearAlgebra.normalize!(::Union{State,Operator}, tn::Product, p::Real)
+    for tensor in tensors(tn)
+        normalize!(tensor, p)
+    end
+    return tn
+end
+
+overlap(a::Product, b::Product) = overlap(socket(a), a, socket(b), b)
```

Comment:
```suggestion
"""
    overlap(a::Product, b::Product)

Compute the overlap between two [`Product`](@ref) ansatzes.
"""
overlap(a::Product, b::Product) = overlap(socket(a), a, socket(b), b)
```

Reply:
does `overlap` conjugate one of the two? I guess it'd be good to specify it in the docstring

Reply:
Thanks! Sure

Reply:
we don't really need a doctoring for this method, but a general docstring for `overlap` which I believe we already have (and if not, it would be another PR)